src/plugins/intel_gpu/src/kernel_selector/cl_kernels/lstm_cell_and_seq_bfyx.cl

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/fetch_data.cl"
+#include "include/batch_headers/common.cl"
+
+#define INPUT0_TYPE_VEC  MAKE_VECTOR_TYPE(INPUT0_TYPE, VEC_SIZE)
+#define INPUT1_TYPE_VEC  MAKE_VECTOR_TYPE(INPUT1_TYPE, VEC_SIZE)
+#define INPUT3_TYPE_VEC  MAKE_VECTOR_TYPE(INPUT3_TYPE, VEC_SIZE)
+#define INPUT4_TYPE_VEC  MAKE_VECTOR_TYPE(INPUT4_TYPE, VEC_SIZE)
+#define OUTPUT_TYPE_VEC  MAKE_VECTOR_TYPE(OUTPUT_TYPE, VEC_SIZE)
+#define READ_VEC(offset, ptr) CAT(vload, VEC_SIZE)(offset, ptr)
+
+#ifdef SEQUENCE
+#define GET_IN0_IDX(b, f, y) INPUT0_GET_INDEX(b, f, y, 0)
+#define GET_IN1_IDX(b, f, y) INPUT1_GET_INDEX(b, f, y, 0)
+#define GET_IN3_IDX(b, f) INPUT3_GET_INDEX(0, b, f, 0)
+#define GET_IN4_IDX(b, f) INPUT4_GET_INDEX(0, b, f, 0)
+#else
+#define GET_IN0_IDX(b, f, y) INPUT0_GET_INDEX(b, y, 0, 0) 
+#define GET_IN1_IDX(b, f, y) INPUT1_GET_INDEX(b, y, 0, 0)
+#define GET_IN3_IDX(b, f) INPUT3_GET_INDEX(b, f, 0, 0)
+#define GET_IN4_IDX(b, f) INPUT4_GET_INDEX(b, f, 0, 0)
+#endif
+
+KERNEL(lstm_cell_and_seq_bfyx)(
+    const __global INPUT0_TYPE* x,
+    const __global INPUT1_TYPE* initial_hidden_state,
+    const __global INPUT2_TYPE* initial_cell_state,
+    const __global INPUT3_TYPE* W,
+    const __global INPUT4_TYPE* R,
+    const __global INPUT5_TYPE* B,
+#ifdef SEQUENCE
+    const __global INPUT6_TYPE* sequence_lengths,
+    __global OUTPUT_TYPE* hidden_history,
+    __global OUTPUT1_TYPE* hidden_state,
+    __global OUTPUT2_TYPE* cell_state
+#else
+    __global OUTPUT_TYPE* hidden_state,
+    __global OUTPUT1_TYPE* cell_state
+#endif
+)
+{
+    const uint b = get_global_id(1);
+    const uint local_idx = get_local_id(0);
+    const uint weight_offsets[4] = {GEMM_OFFSET_F, GEMM_OFFSET_I, GEMM_OFFSET_Z, GEMM_OFFSET_O};
+    #ifdef SEQUENCE
+        const uint real_seq_length = sequence_lengths[INPUT6_GET_INDEX(b, 0, 0, 0)];
+    #else
+        const uint real_seq_length = 1;
+    #endif
+
+    unroll_for(uint i=0;i<real_seq_length;++i){
+        #ifdef SEQUENCE
+            #if DIRECTION == 1 //reverse
+                const uint prev_idx = real_seq_length - i;
+            #else
+                const uint prev_idx = i-1;
+            #endif
```

Comment:
[random spot] As I can see, bidirectional LSTM sequence works incorrectly.
```
[ RUN      ] LSTMSequenceCommonZeroClip/LSTMSequenceTest.Inference/mode=PURE_SEQ_seq_lengths=2_batch=10_hidden_size=10_input_size=10_IS=(10.10)(10.10)(10.10)(40.10)(40.10)(40)_activations=(sigmoid.tanh.tanh)_direction=bidirectional_clip=0_WRBType=CONSTANT_modelType=f32_targetDevice=GPU_
Expected: 0.995055 Actual: 0 Coordinate: 20 Diff: 0.995055 calculated_abs_threshold: 9.96247e-05 abs_threshold: 1.19209e-07 rel_threshold: 0.0001
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:97: Failure
[ COMPARATION ] COMPARATION IS FAILED! incorrect elem counter: 1 among 400 shapes.
[  FAILED  ] LSTMSequenceCommonZeroClip/LSTMSequenceTest.Inference/mode=PURE_SEQ_seq_lengths=2_batch=10_hidden_size=10_input_size=10_IS=(10.10)(10.10)(10.10)(40.10)(40.10)(40)_activations=(sigmoid.tanh.tanh)_direction=bidirectional_clip=0_WRBType=CONSTANT_modelType=f32_targetDevice=GPU_, where GetParam() = (PURE_SEQ, 2, 10, 10, 10, { "sigmoid", "tanh", "tanh" }, 0, bidirectional, CONSTANT, f32, "GPU") (52 ms)
```
(I've manually disabled bidirectional sequence decomposition and enforced usage of lstm_seq primitive to get that resul)t

Reply:
done